design/object-store-realm.md

```diff
@@ -0,0 +1,214 @@
+A master zone needs to be created for secondary zones to pull from to replicate across zones
+```
+
+## Object Store Multisite Walkthrough
+
+### Creating a Realm & Zonegroups
+
+#### Config
+
+When the storage admin is ready to create a realm and zonegroups for object storage, the admin will specify his/her desired settings in the Realm section on the configuration file.
+
+This example `object-store-realm.yaml`, names a realm with 2 zonegroups.
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStoreRealm
+metadata:
+  name: my-realm
+  namespace: rook-ceph
+spec:
+  bootstrap-realm:
+    name: my-realm
+    zonegroups:
+      my-zonegroup-a:
+        is-master: true
+        master-zone: zone-z
+      my-zonegroup-b:
+        is-master: false
+        master-zone: zone-y
+```
+
+Now create the object store realm.
+```bash
+kubectl create -f object-store-realm.yaml
+```
+
+#### Steps
+
+At this point the Rook operator recognizes that a new object store realm resource needs to be configured. The operator will create all of the resources to start the object store realm.
+
+1. A Ceph realm is created if it does not exist already
+2. A Ceph master zonegroup is created in the realm if it does not exist already
+3. Any non-master Ceph zonegroup is created in the realm if it does not exist already
+4. A master zone is created for the zonegroup.
+
+After these steps the admin should start up object stores configured with the same zone, zonegroup, and realm as configured (see next section).
+
+#### Notes
+1. The zonegroups section of the object-store-realm CR is only additive. If the playbook is reapplied and zonegroup already exist the zonegroup won't be recreated. If a zonegroup previously created is removed from the list of zonegroups, it will not be deleted. If a new zonegroup is added to the list it will be created. 
+2. The same principle applies to master-zones.
+3. When resource is deleted, realm and zonegroups are not deleted from the cluster. Realm and zonegroup deletion must be done through toolboxes.
+
+### Creating a Zone on the same cluster the object-store-realm was created.
+
+Once the storage admin creates a realm and zonegroup(s) on a cluster, the admin is ready to add object-stores to a master zone on a realm or add another zone to the zonegroup. 
+The admin will specify his/her configuration settings in the realm section of the object-store like in the following `object-store.yaml`.
+
+All objectstores with the realm section must be named in the form `realm_name` + `zone_name`.`zone_name` can be the name of a master zone created in the object store realm CR or the name of any other one..
+
+If the name of the objectstore is `realm_name` + '-' + `master_zone_name`, the the object stores created will be added to the master zone.
+
+If the name of the objectstore is not `realm_name` + '-' + `master_zone_name`, then a new zone will be created with the object stores created via the resource.
```

Comment:
why do we need these naming conventions? why not specifying the realm and zone in the object store?